### PR TITLE
Rebalance Troubadour & Trouvere 20221221

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: https://github.com/rathena/rathena/issues/8018

* **Server Mode**: renewal

* **Description of Pull Request**: 

Troubadour & Trouvere
---------------------------------

1. [Rhythm Shooting](https://www.divine-pride.net/database/skill/5355)
	- Increases cooldown from 0.15 seconds to 0.35 seconds.
	- Increases base damage from 800%/1400%Atk to 3700%/4550%Atk per hit based on level 5.
	- Increases factor weight of CON in skill formula from 3 to 5 (7 on targets marked with sound brand).

2. [Metallic Fury](https://www.divine-pride.net/database/skill/5356)
	- Reworks skill to dealing damage to surrounding enemies around the target (9 x 9 cells based on level 5).
	- Increases cooldown from 0.3 seconds to 0.4 seconds.
	- Increases global cool time from 0.3 seconds to 0.5 seconds.
	- No longer remove sound brand debuff from target. (doesn't mention it changelog but it works)
	- Increases base damage from 11000%Matk to 13000%/18000%(targets marked with sound brand)Matk based on level 5.
	- Increases factor weight of SPL in skill formula (on targets marked with sound brand) from 1.5x of Stage Manner skill level to 2x of Stage Manner skill level. 

Additionally
-----------------------------
Rhythm Shooting
- No longer remove sound brand debuff from target. (from skill description)

### CODE REVIEW

File: src/map/battle.cpp [hunk 1]
```diff
@@ -5734,13 +5734,15 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
...
+			skillratio += -100 + 450 + 650 * skill_lv;
 
 			if (sd && pc_checkskill(sd, TR_STAGE_MANNER) > 0)
-				skillratio += 3 * sstatus->con;
+				skillratio += 5 * sstatus->con;
 
-			if (tsc && tsc->getSCE(SC_SOUNDBLEND))
-				skillratio += 100 + 100 * skill_lv;
+			if (tsc && tsc->getSCE(SC_SOUNDBLEND)) {
+				skillratio += 350 + 100 * skill_lv;
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:5743]: Not sure if this is a typo in the skill description, but level 4 does not match the formula.
Pretty sure that they have a typo though.

File: src/map/skill.cpp [hunk 2]
```diff
@@ -2183,8 +2183,6 @@ int skill_additional_effect( struct block_list* src, struct block_list *bl, uint
 		[[fallthrough]];
 	case WM_METALICSOUND:
 	case WM_REVERBERATION:
-	case TR_RHYTHMSHOOTING:
```
Comment (Lemongrass3110) [inline src/map/skill.cpp:2186]: Not mentioned in the changelog?
If you are sure about this, just ignore this comment.

File: src/map/battle.cpp [hunk 3]
```diff
@@ -8037,12 +8039,13 @@ struct Damage battle_calc_magic_attack(struct block_list *src,struct block_list
...
-							int count = map_forcountinarea(skill_check_bl_sc,target->m,target->x - area,target->y - area,target->x + area,target->y + area,5,BL_MOB,SC_SOUNDBLEND);
-							skillratio += -100 + (2200 + 300 * count) * skill_lv + 5 * sstatus->spl;
-							RE_LVL_DMOD(100);
-						}
+					case TR_METALIC_FURY:
+						skillratio += -100 + 2600 * skill_lv;
+						if (tsc && tsc->getSCE(SC_SOUNDBLEND))
+							skillratio += 1000 * skill_lv;
+						if (sd)
+							skillratio += 2 * pc_checkskill(sd, TR_STAGE_MANNER) * sstatus->spl;
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:8047]:
```suggestion
						if (tsc && tsc->getSCE(SC_SOUNDBLEND)){
							skillratio += 1000 * skill_lv;
							skillratio += 2 * pc_checkskill(sd, TR_STAGE_MANNER) * sstatus->spl;
						}
```
The description (of Sigma) sounds like this should only apply to targets that are marked with SC_SOUNDBLEND.

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/battle.cpp (hunk 3 above)
Comment (datawulf) [reply src/map/battle.cpp:8047]: Before it should have been 1.5 x SPL against any target. Now I read it as 1.5 x SPL without SC_SOUNDBLEND and 2 x SPL with SC_SOUNDBLEND

File: src/map/battle.cpp (hunk 1 above)
Comment (Atemo) [reply src/map/battle.cpp:5743]: Possible. I guess we should follow the official value until we have more information.

File: src/map/skill.cpp (hunk 2 above)
Comment (Atemo) [reply src/map/skill.cpp:2186]: This is not mentioned in the official changelog (like `TR_METALIC_FURY`). But the skill description has been changed as well

File: src/map/battle.cpp (hunk 3 above)
Comment (Atemo) [reply src/map/battle.cpp:8047]: Possible, the part without SC_SOUNDBLEND is not clear.

**Pull request merged**